### PR TITLE
Clean up *OverlappedAsyncResults types.

### PR DESCRIPTION
**Pull request by pgavlin**

### CODE REVIEW

**Comment by pgavlin** [comment]
@davidsh, @stephentoub PTAL

**Comment by stephentoub** [comment]
LGTM

**Comment by davidsh** [comment]
Left minor comments.  Otherwise LGTM.

File: src/System.Net.Sockets/src/System/Net/Sockets/_AcceptOverlappedAsyncResult.Windows.cs [hunk 1]
```diff
@@ -71,7 +61,10 @@ out remoteSocketAddress.InternalSize
                         ref handle,
                         Marshal.SizeOf(handle));
 
-                    if (errorCode == SocketError.SocketError) errorCode = (SocketError)Marshal.GetLastWin32Error();
+                    if (errorCode == SocketError.SocketError)
+                    {
+                        errorCode = (SocketError)Marshal.GetLastWin32Error();
+                    }
```
Comment (CIPop) [inline src/System.Net.Sockets/src/System/Net/Sockets/_AcceptOverlappedAsyncResult.Windows.cs:67]: Nit: (coding style) new line after '}'

File: src/System.Net.Sockets/src/System/Net/Sockets/_BaseOverlappedAsyncResult.Unix.cs [hunk 2]
```diff
@@ -10,18 +10,12 @@
 
 namespace System.Net.Sockets
 {
+    // BaseOverlappedAsyncResult
```
Comment (CIPop) [inline src/System.Net.Sockets/src/System/Net/Sockets/_BaseOverlappedAsyncResult.Unix.cs:13]: Nit: I think we should remove the line above: it's just the name of the class below.

File: src/System.Net.Sockets/src/System/Net/Sockets/_BaseOverlappedAsyncResult.Windows.cs [hunk 3]
```diff
@@ -10,30 +10,18 @@
 
 namespace System.Net.Sockets
 {
+    // BaseOverlappedAsyncResult
```
Comment (CIPop) [inline src/System.Net.Sockets/src/System/Net/Sockets/_BaseOverlappedAsyncResult.Windows.cs:13]: Same as above. (Also, other occurrences throughout the file.)

File: src/System.Net.Sockets/src/System/Net/Sockets/_BaseOverlappedAsyncResult.Windows.cs [hunk 4]
```diff
@@ -169,23 +150,18 @@ private unsafe static void CompletionPortCallback(uint errorCode, uint numBytes,
...
         // The following property returns the Win32 unsafe pointer to
         // whichever Overlapped structure we're using for IO.
-        //
         internal SafeHandle OverlappedHandle
         {
             get
             {
-                //
-                // on WinNT we need to use (due to the current implementation)
+                // On WinNT we need to use (due to the current implementation)
```
Comment (CIPop) [inline src/System.Net.Sockets/src/System/Net/Sockets/_BaseOverlappedAsyncResult.Windows.cs:159]: NIT: Windows instead of WinNT.

**Comment by CIPop** [comment]
LGTM (A few nitpicks on code style.)

**Pull request merged**